namui/src/namui/skia/base/paint_builder.rs

```diff
@@ -19,10 +18,22 @@ pub struct PaintBuilder {
     stroke_cap: Option<StrokeCap>,
     stroke_join: Option<StrokeJoin>,
     color_filter: Option<(Color, BlendMode)>,
-    shader: Option<IntermediateShader>,
+    shader: Option<Arc<Shader>>,
 }
 
-static PAINT_CACHE: OnceCell<Mutex<lru::LruCache<PaintBuilder, Arc<Paint>>>> = OnceCell::new();
+#[derive(Clone, PartialEq)]
+pub struct PaintBuilderWithoutShader {
+    color: Option<Color>,
+    paint_style: Option<PaintStyle>,
+    anti_alias: Option<bool>,
+    stroke_width: Option<Px>,
+    stroke_cap: Option<StrokeCap>,
+    stroke_join: Option<StrokeJoin>,
+    color_filter: Option<(Color, BlendMode)>,
+}
+
+static PAINT_CACHE: OnceCell<Mutex<lru::LruCache<PaintBuilderWithoutShader, Arc<Paint>>>> =
+    OnceCell::new();
```

Comment:
I changed paint cache strategy.

paint won't cache using shader. paint just set shader everytime when they pop from cache. If the shader is the same with previous, they don't set shader.

Reply:
This is because shader could be changed too often against other paint properties. for example, when I make scroll, it create shader with time in every frame.

Later I will prevent creating shader in render stage, but only in draw stage. they don't have to create shader in rendering stage.

Reply:
Issue tracked in https://github.com/NamseEnt/namseent/issues/282